### PR TITLE
feat(css): add resource CSS es loadbalancer config

### PR DESCRIPTION
**Pull request by luoping-12345**
**What this PR does / why we need it**:
add resource CSS es loadbalancer config

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```
add resource CSS es loadbalancer config
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/css' TESTARGS='-run TestAccCssEsLoadbalancerConfig_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/css -v -run TestAccCssEsLoadbalancerConfig_basic -timeout 360m -parallel 4
=== RUN   TestAccCssEsLoadbalancerConfig_basic
=== PAUSE TestAccCssEsLoadbalancerConfig_basic
=== CONT  TestAccCssEsLoadbalancerConfig_basic
--- PASS: TestAccCssEsLoadbalancerConfig_basic (963.75s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/css       963.812s
```

### CODE REVIEW

File: huaweicloud/services/css/resource_huaweicloud_css_es_loadbalancer_config.go [hunk 1]
```diff
@@ -0,0 +1,500 @@
...
+		Pending:      []string{"PENDING"},
+		Target:       []string{"FINISHED"},
+		Refresh:      ltCreateRstStateRefreshFunc(d.Id(), client),
+		Timeout:      d.Timeout(schema.TimeoutCreate),
+		Delay:        10 * time.Second,
+		PollInterval: 10 * time.Second,
+	}
+	_, err = stateConf.WaitForStateContext(ctx)
+	if err != nil {
+		return diag.Errorf("error waiting for the cluster createing es-listener to finish: %s", err)
```
Comment (houpeng80) [inline huaweicloud/services/css/resource_huaweicloud_css_es_loadbalancer_config.go]:
```suggestion
		return diag.Errorf("error waiting for the cluster creating es-listener to finish: %s", err)
```

File: huaweicloud/services/css/resource_huaweicloud_css_es_loadbalancer_config.go [hunk 2]
```diff
@@ -0,0 +1,500 @@
...
+	}
+
+	return resourceEsLoadbalancerConfigRead(ctx, d, meta)
+}
+
+func buildCreateEsListenerBodyParams(d *schema.ResourceData) map[string]interface{} {
+	bodyParams := map[string]interface{}{
+		"protocol_port":  d.Get("protocol_port").(int),
+		"server_cert_id": d.Get("server_cert_id"),
+		"ca_cert_id":     d.Get("ca_cert_id"),
```
Comment (houpeng80) [inline huaweicloud/services/css/resource_huaweicloud_css_es_loadbalancer_config.go]: these two params should be ignore if the values is nil

File: huaweicloud/services/css/resource_huaweicloud_css_es_loadbalancer_config.go [hunk 3]
```diff
@@ -0,0 +1,500 @@
...
+}
+
+func ltCreateRstStateRefreshFunc(clusterID string, client *golangsdk.ServiceClient) resource.StateRefreshFunc {
+	return func() (interface{}, string, error) {
+		resp, err := getEsListenerDetail(clusterID, client)
+		if err != nil {
+			return nil, "ERROR", err
+		}
+
+		listencer := utils.PathSearch("listener", resp, nil)
```
Comment (houpeng80) [inline huaweicloud/services/css/resource_huaweicloud_css_es_loadbalancer_config.go]:
```suggestion
		listener := utils.PathSearch("listener", resp, nil)
```

File: huaweicloud/services/css/resource_huaweicloud_css_es_loadbalancer_config.go [hunk 4]
```diff
@@ -0,0 +1,500 @@
...
+	createEsListenerHttpUrl := "v1.0/{project_id}/clusters/{cluster_id}/es-listeners"
+	createEsListenerPath := client.Endpoint + createEsListenerHttpUrl
+	createEsListenerPath = strings.ReplaceAll(createEsListenerPath, "{project_id}", client.ProjectID)
+	createEsListenerPath = strings.ReplaceAll(createEsListenerPath, "{cluster_id}", clusterId)
+
+	createEsListenerOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		OkCodes: []int{
+			200,
+		},
```
Comment (houpeng80) [inline huaweicloud/services/css/resource_huaweicloud_css_es_loadbalancer_config.go]: the defalut ok codes contains 200, it is better to delete this

File: docs/resources/css_es_loadbalancer_config.md [hunk 5]
```diff
@@ -0,0 +1,127 @@
+---
+subcategory: "Cloud Search Service (CSS)"
+---
```
Comment (houpeng80) [inline docs/resources/css_es_loadbalancer_config.md]: add more information refer to data source

File: docs/resources/css_es_loadbalancer_config.md [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+
+# huaweicloud_css_es_loadbalancer_config
+
+Manages CSS loadbalancer resource within HuaweiCloud.
```
Comment (houpeng80) [inline docs/resources/css_es_loadbalancer_config.md]:
```suggestion
Manages CSS ElasticSearch loadbalancer resource within HuaweiCloud.
```

File: docs/resources/css_es_loadbalancer_config.md [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+
+In addition to all arguments above, the following attributes are exported:
+
+* `id` - The resource ID.
+
+* `server_cert_name` - The server certificate name.
+
+* `ca_cert_name` - The CA certificate name.
+
+* `elb_enabled` - Whether open the ELB loadbalancer.
```
Comment (houpeng80) [inline docs/resources/css_es_loadbalancer_config.md]:
```suggestion
* `elb_enabled` - Whether the loadbalancer is enabled.
```

File: docs/resources/css_es_loadbalancer_config.md [hunk 8]
```diff
@@ -0,0 +1,127 @@
...
+  If omitted, the provider-level region will be used.
+  Changing this creates a new resource.
+
+* `cluster_id` - (Required, String, ForceNew) Specifies the ID of the CSS cluster.
+  Changing this creates a new resource.
+
+* `agency` - (Required, String, ForceNew) Specifies the IAM agency used to access ELB.
+  Changing this creates a new resource.
+
+* `loadbalancer_id` - (Required, String, ForceNew) Specifies the ID of ELB loadbalancer.
```
Comment (houpeng80) [inline docs/resources/css_es_loadbalancer_config.md]:
```suggestion
* `loadbalancer_id` - (Required, String, ForceNew) Specifies the ID of loadbalancer.
```
and fix other places

File: docs/resources/css_es_loadbalancer_config.md [hunk 9]
```diff
@@ -0,0 +1,127 @@
...
+
+* `protocol_port` - Backend server business port number.
+
+* `status` - The health status of the backend cloud server.
+
+## Timeouts
+
+This resource provides the following timeouts configuration options:
+
+* `create` - Default is 3 minutes.
```
Comment (houpeng80) [inline docs/resources/css_es_loadbalancer_config.md:123]: the delete timeout should be added

File: docs/resources/css_es_loadbalancer_config.md [hunk 10]
```diff
@@ -0,0 +1,127 @@
...
+
+## Import
+
+The xxx can be imported using `cluster_id`, e.g.
+
+```bash
+$ terraform import huaweicloud_css_loadbalancer.test <cluster_id>
+```
```
Comment (houpeng80) [inline docs/resources/css_es_loadbalancer_config.md]:
```suggestion
The CSS ElasticSearch loadbalancer config can be imported using `cluster_id`, e.g.

```bash
$ terraform import huaweicloud_css_es_loadbalancer_config.test <cluster_id>
```
```

File: huaweicloud/services/acceptance/css/resource_huaweicloud_css_es_loadbalancer_config_test.go [hunk 11]
```diff
@@ -0,0 +1,223 @@
...
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+)
+
+func getCssEsLoadbalancerConfigFunc(conf *config.Config, state *terraform.ResourceState) (interface{}, error) {
+	client, err := conf.NewServiceClient("css", acceptance.HW_REGION_NAME)
+	if err != nil {
+		return nil, fmt.Errorf("error creating CSS v1 client: %s", err)
```
Comment (houpeng80) [inline huaweicloud/services/acceptance/css/resource_huaweicloud_css_es_loadbalancer_config_test.go]:
```suggestion
		return nil, fmt.Errorf("error creating CSS client: %s", err)
```
keep consistent with the resource

File: huaweicloud/services/acceptance/css/resource_huaweicloud_css_es_loadbalancer_config_test.go [hunk 12]
```diff
@@ -0,0 +1,223 @@
...
+		ProviderFactories: acceptance.TestAccProviderFactories,
+		CheckDestroy:      rc.CheckResourceDestroy(),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCssEsLoadbalancerConfig_basic(rName),
+				Check: resource.ComposeTestCheckFunc(
+					rc.CheckResourceExists(),
+					resource.TestCheckResourceAttrPair(resourceName, "loadbalancer_id",
+						"huaweicloud_elb_loadbalancer.test", "id"),
+					resource.TestCheckResourceAttr(resourceName, "agency", "css_elb_agency"),
```
Comment (houpeng80) [inline huaweicloud/services/acceptance/css/resource_huaweicloud_css_es_loadbalancer_config_test.go]: how about add this agency to accentence as a variable, if it is deleted, then an error will be occured

File: huaweicloud/services/acceptance/css/resource_huaweicloud_css_es_loadbalancer_config_test.go [hunk 13]
```diff
@@ -0,0 +1,223 @@
...
+
+func testAccCssEsLoadbalancerConfig_basic(rName string) string {
+	return fmt.Sprintf(`
+%[1]s
+
+resource "huaweicloud_css_es_loadbalancer_config" "test" {
+  cluster_id = huaweicloud_css_cluster.test.id
+  agency = "css_elb_agency"
+  loadbalancer_id = huaweicloud_elb_loadbalancer.test.id
+  protocol_port = 9200
```
Comment (houpeng80) [inline huaweicloud/services/acceptance/css/resource_huaweicloud_css_es_loadbalancer_config_test.go]: align the equal signs, and fix other places

File: huaweicloud/services/css/resource_huaweicloud_css_es_loadbalancer_config.go [hunk 14]
```diff
@@ -0,0 +1,494 @@
...
+
+	createEsListenerOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		MoreHeaders:      map[string]string{"Content-Type": "application/json"},
+	}
+
+	createEsListenerOpt.JSONBody = utils.RemoveNil(buildCreateEsListenerBodyParams(d))
+	_, err = client.Request("POST", createEsListenerPath, &createEsListenerOpt)
+	if err != nil {
+		return diag.Errorf("error create CSS es-listener: %s", err)
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/css/resource_huaweicloud_css_es_loadbalancer_config.go]:
```suggestion
		return diag.Errorf("error creating CSS es-listener: %s", err)
```

File: huaweicloud/services/css/resource_huaweicloud_css_es_loadbalancer_config.go [hunk 15]
```diff
@@ -0,0 +1,494 @@
...
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+	client, err := cfg.NewServiceClient("css", region)
+	if err != nil {
+		return diag.Errorf("error creating CSS client: %s", err)
+	}
+
+	getEsListenerRespBody, err := getEsListenerDetail(d.Id(), client)
+	if err != nil {
+		return common.CheckDeletedDiag(d, err, "error get CSS es-listener")
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/css/resource_huaweicloud_css_es_loadbalancer_config.go]:
```suggestion
		return common.CheckDeletedDiag(d, err, "error getting CSS es-listener")
```

File: huaweicloud/services/css/resource_huaweicloud_css_es_loadbalancer_config.go [hunk 16]
```diff
@@ -0,0 +1,494 @@
...
+
+	createEsListenerOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		MoreHeaders:      map[string]string{"Content-Type": "application/json"},
+	}
+
+	createEsListenerOpt.JSONBody = utils.RemoveNil(buildUpdateEsListenerBodyParams(d))
+	_, err = client.Request("PUT", createEsListenerPath, &createEsListenerOpt)
+	if err != nil {
+		return diag.Errorf("error update CSS es-listener: %s", err)
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/css/resource_huaweicloud_css_es_loadbalancer_config.go]:
```suggestion
		return diag.Errorf("error updating CSS es-listener: %s", err)
```

File: huaweicloud/services/css/resource_huaweicloud_css_es_loadbalancer_config.go [hunk 17]
```diff
@@ -0,0 +1,494 @@
...
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+	client, err := cfg.NewServiceClient("css", region)
+	if err != nil {
+		return diag.Errorf("error creating CSS client: %s", err)
+	}
+
+	err = openOrCloseLoadbalancer(client, d, false)
+	if err != nil {
+		return diag.Errorf("error close CSS loadbalancer: %s", err)
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/css/resource_huaweicloud_css_es_loadbalancer_config.go]:
```suggestion
		return diag.Errorf("error closing CSS loadbalancer: %s", err)
```

**Pull request merged**